### PR TITLE
Fix authentication error payload

### PR DESCRIPTION
**Pull request by PaulAsjes**
## Description
Exceptions were only returning specific hardcoded values, which isn't accurate any more since an error can have a plethora of parameters in it. 

This fixes it by dumping the contents of the API response into the error object, giving the user all the information they need to fix the error.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: tests/test_sync_http_client.py [hunk 1]
```diff
@@ -222,29 +222,11 @@ def test_bad_request_exceptions_include_expected_request_data(self):
         except BadRequestException as ex:
             assert (
                 str(ex)
-                == "(message=No message, request_id=request-123, error=example_error, error_description=Example error description)"
+                == "(message=No message, error=example_error, error_description=Example error description, foo=bar, request_id=request-123)"
             )
         except Exception as ex:
             assert ex.__class__ == BadRequestException
 
-    def test_bad_request_exceptions_exclude_expected_request_data(self):
```
Comment (PaulAsjes) [inline tests/test_sync_http_client.py:230]: What we want is for everything in the return value to be added to exceptions, this test does the opposite.

File: workos/exceptions.py [hunk 2]
```diff
@@ -8,70 +8,35 @@ class BaseRequestException(Exception):
...
-            self.error_description = None
-            self.pending_authentication_token = None
-
-        headers = response.headers
-        self.request_id = headers.get("X-Request-ID")
-
-    def __str__(self) -> str:
-        message = self.message or "No message"
-        exception = "(message=%s" % message
+        self.message = self.extractFromJson("message", "No message")
```
Comment (gcarvelli) [inline workos/exceptions.py:18]: This is 100x nicer than `except ValueError`, good stuff

File: workos/exceptions.py [hunk 3]
```diff
@@ -8,70 +8,35 @@ class BaseRequestException(Exception):
...
+        self.error = self.extractFromJson("error", None)
+        self.errors = self.extractFromJson("errors", None)
+        self.code = self.extractFromJson("code", None)
+        self.error_description = self.extractFromJson("error_description", None)
```
Comment (gcarvelli) [inline workos/exceptions.py:22]: `error` and `error_description` previously defaulted to `"Unknown"` so this is a small change in behavior ([line](https://github.com/workos/workos-python/pull/360/files#diff-fad3453bb4c7313cf5431a3e0e2939488c284ef8cb50b2f250ed77d4fd8035f2L108))

**Review by gcarvelli — APPROVED** [review]

File: workos/exceptions.py (hunk 3 above)
Comment (PaulAsjes) [reply workos/exceptions.py:22]: Oh good catch!

File: workos/exceptions.py (hunk 2 above)
Comment (PaulAsjes) [reply workos/exceptions.py]: Thanks! Also lol snake case

**Comment by azuline** [comment]
thank you!

**Pull request merged**